data/schema/mapper.go

```diff
@@ -10,47 +11,145 @@ import (
 	"github.com/tidepool-org/platform/data/types/basal/scheduled"
 	"github.com/tidepool-org/platform/data/types/basal/temporary"
 	"github.com/tidepool-org/platform/data/types/blood/glucose/continuous"
+	"github.com/tidepool-org/platform/data/types/bolus/biphasic"
+	"github.com/tidepool-org/platform/data/types/bolus/normal"
 	"github.com/tidepool-org/platform/errors"
 )
 
 const ErrEventTime = "unable to parse event time"
 
-func (s *BasalSample) mapForBasal(event *basal.Basal) error {
+func (s *BolusSample) MapForNormalBolus(event *normal.Normal) error {
 	var err error
 	// assign an uuid for keeping a link between the two collection
-	event.InternalID = uuid.New().String()
+	s.Uuid = *event.ID
+
+	if event.GUID != nil {
+		s.Guid = *event.GUID
+	}
+
+	if event.DeviceID != nil {
+		s.DeviceId = *event.DeviceID
+	}
+	// bolus struct field
+	s.BolusType = "normal"
+	if event.InsulinOnBoard != nil {
+		if event.InsulinOnBoard.InsulinOnBoard != nil {
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏Avoid to stack ifs. Something like that:
`if event.InsulinOnBoard != nil && event.InsulinOnBoard.InsulinOnBoard != nil {`